src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -44,14 +46,16 @@ public function __construct(FilesystemInterface $filesystem, int $segments)
 
         $this->endpoint = (string) $this->adapter->getClient()->getEndpoint();
         $this->bucketName = $this->adapter->getBucket();
+
+        $this->url = $url;
     }
 
     public function getPath(array $storageOptions): string
     {
         $filePath = $this->getFilePath($storageOptions);
         $path = $this->adapter->applyPathPrefix($filePath);
 
-        return $this->endpoint . '/' . $this->bucketName . '/' . \ltrim($path, '/');
+        return ($this->url ?? $this->endpoint . '/' . $this->bucketName) . '/' . \ltrim($path, '/');
```

Comment:
Can be simplified as arleady handled in the construct:

```suggestion
        return $this->url . '/' . $this->bucketName . '/' . \ltrim($path, '/');
```